recipes/src/main/java/org/apache/camel/quarkus/update/RemovedComponentsRecipe.java

```diff
@@ -0,0 +1,127 @@
+        { "camel-quarkus-microprofile-metrics", "camel-quarkus-opentelemetry", "camel-quarkus-micrometer" },
+        { "camel-quarkus-opentracing", "camel-quarkus-opentelemetry", "camel-quarkus-micrometer" },
+        { "camel-quarkus-rabbitmq", "camel-quarkus-spring-rabbitmq???" },
+        { "camel-quarkus-xstream", "camel-quarkus-jacksonxml" },
+    }).collect(Collectors.toMap(data -> data[0], data -> Arrays.asList(Arrays.copyOfRange(data, 1, data.length))));
+
+    private static Set<String> TO_BE_REINTRODUCED  = new HashSet<>(Arrays.asList(
+            "camel-quarkus-datasonnet",
+            "camel-quarkus-smallrye-reactive-messaging",
+            "camel-quarkus-tika",
+            "camel-quarkus-xmlsecurity"));
+
+    private final static XPathMatcher DEPENDENC_MATCHER = new XPathMatcher("//dependencies/dependency");
+
+    @Override
+    public String getDisplayName() {
+        return "Removed Camel components.";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Removed Camel components.";
+    }
+
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new MavenVisitor<>() {
+
+            @Override
+            public Xml visitTag(Xml.Tag tag, ExecutionContext ctx) {
+                Xml.Tag t = (Xml.Tag) super.visitTag(tag, ctx);
+
+                if(!DEPENDENC_MATCHER.matches(getCursor())) {
+                    return t;
+                }
+
+                if (GROUP_ID.equals(t.getChildValue("groupId").orElse(null))) {
+                    String artifactdD = t.getChildValue("artifactId").orElse(null);
+                    if(artifactdD != null && ARTIFACT_IDS.contains(artifactdD)) {
+                        //add a comment to the artifactId
+                        LinkedList l = new LinkedList(t.getContent());
+                        l.addFirst(RecipesUtil.createXmlComment(commentToRemovedArtifactId(artifactdD, t.toString())));
+
+                        return RecipesUtil.createXmlComment(commentToRemovedArtifactId(artifactdD, t.print(getCursor()))).withPrefix(t.getPrefix());
+                    }
+                }
+                return t;
+            }
+
+            @Override
+            protected void doAfterVisit(Recipe recipe) {
+                super.doAfterVisit(recipe);
+            }
+
+            private String commentToRemovedArtifactId(String artifactId, String theDefinition) {
+                if (ALTERNATIVE_COMPONENTS.containsKey(artifactId)) {
+                    return String.format("Extension %s was removed, consider %s instead. %s", artifactId, ALTERNATIVE_COMPONENTS.get(artifactId).stream().collect(Collectors.joining(" or ")), theDefinition);
+                } else if (TO_BE_REINTRODUCED.contains(artifactId)) {
+                    return String.format("Extension %s was removed, but should be reintroduced. %s", artifactId, theDefinition);
```

Comment:
Wonder if we need a `reintroduced` concept? Once CQ 3.0.0 is GA, I'd assume that we'd have added back anything that we considered important.